### PR TITLE
SAK-40148 Assignments conversion for group assignments with SITE access

### PR DESCRIPTION
**Pull request by juanjmerono**
…s type SITE

### CODE REVIEW

File: assignment/impl/src/java/org/sakaiproject/assignment/impl/conversion/AssignmentConversionServiceImpl.java [hunk 1]
```diff
@@ -435,9 +435,26 @@ private AssignmentSubmission submissionReintegration(Assignment assignment, O11S
...
-            		&& assignment.getGroups().contains("/site/"+assignment.getContext()+"/group/"+submission.getSubmitterid())) {
-                s.setGroupId(submission.getSubmitterid());
+            if (StringUtils.isNotBlank(submission.getSubmitterid())) {
+            	try {
+            		// The id must be any group if is SITE mode or a some of the accesed groups if its GROUPED
+            		if (Assignment.Access.GROUP.equals(assignment.getTypeOfAccess())) {
+            			if (assignment.getGroups().contains("/site/"+assignment.getContext()+"/group/"+submission.getSubmitterid())) {
+            				s.setGroupId(submission.getSubmitterid());
+            			} else {
+            				return null;
```
Comment (ern) [inline assignment/impl/src/java/org/sakaiproject/assignment/impl/conversion/AssignmentConversionServiceImpl.java]: returning null will not import the submission is this expected?

File: assignment/impl/src/java/org/sakaiproject/assignment/impl/conversion/AssignmentConversionServiceImpl.java [hunk 2]
```diff
@@ -435,9 +435,26 @@ private AssignmentSubmission submissionReintegration(Assignment assignment, O11S
...
+            			}
+            		} else {
+            			Site assignmentSite = siteService.getSite(assignment.getContext());
+            			if (assignmentSite.getGroup(submission.getSubmitterid())!=null) {
+            				s.setGroupId(submission.getSubmitterid());
+            			} else {
+            				return null;
```
Comment (ern) [inline assignment/impl/src/java/org/sakaiproject/assignment/impl/conversion/AssignmentConversionServiceImpl.java]: Same?

**Comment by jonespm** [comment]
The only thing we were talking about on the call is the method siteGroupAssignmentConversion not asserting anything (because the methods are asserting)

I guess the quick fix for this would be to put at the end of this method

`assertTrue(Boolean.TRUE);`

As if it gets to the end without throwing an exception it's has passed. This is probably more in line with @ern  idea of keeping the other methods only checking without a return.

**Comment by juanjmerono** [comment]
I've realized that this situation is due to a problem in Sakai 11 [SAK-40028](https://jira.sakaiproject.org/browse/SAK-40028), that create inconsistent data. I mean is not possible having assignments with TYPE: Group and ACCESS_TYPE: Site, so I guess the solution must be change the ACCESS_TYPE and add submissions and groups to the assignment.

**Comment by juanjmerono** [comment]
I've changed the purpose of this fix, now the conversion script is trying to recover the corrupted data in previous instances. If detects some group assignment with SITE as type of access then changes the access type and recover the groups from the exiting submissions.

**Pull request merged**